multilep/interface/LeptonAnalyzer.h

```diff
@@ -105,18 +105,40 @@ class LeptonAnalyzer {
     double _lMuonTrackPt[nL_max];
     double _lMuonTrackPtErr[nL_max];
 
-    //tau ID decisions
-    bool _tauMuonVeto[nL_max];
-    bool _tauEleVeto[nL_max];
-    bool _decayModeFindingNew[nL_max];
-    bool _tauVLooseMvaNew[nL_max]; //"old tau id's will be stored in the POG id definitions (vloose := veto), however very tight is stored separately
-    bool _tauLooseMvaNew[nL_max];
-    bool _tauMediumMvaNew[nL_max];
-    bool _tauTightMvaNew[nL_max];
-    bool _tauVTightMvaNew[nL_max];
-    bool _tauVTightMvaOld[nL_max];
-
-    //raw tau id values
+    bool _tauMuonVetoLoose[nL_max];                                                                       //tau specific variables
+    bool _tauMuonVetoTight[nL_max];                                                                       
+    bool _tauEleVetoVLoose[nL_max];
+    bool _tauEleVetoLoose[nL_max];
+    bool _tauEleVetoMedium[nL_max];
+    bool _tauEleVetoTight[nL_max];
+    bool _tauEleVetoVTight[nL_max];
+    bool _decayModeFinding[nL_max];                      
+    unsigned int _tauDecayMode[nL_max];                                                         // As in https://twiki.cern.ch/twiki/bin/viewauth/CMS/TauIDRecommendation13TeV#Decay_Mode_Reconstruction 
+    unsigned int _tauGenStatus[nL_max];                                                         //1: prompt ele, 2:prompt mu, 3: ele from leptonic tau, 4:mu from leptonic tau, 5: hadronically decayed tau, 6:rest 
+    bool _tauPOGVLoose2015[nL_max];                                                             //version of ID to use in MiniAOD: MC 80X_mcRun2_asymptotic_2016_TrancheIV_v6, Data 03Feb2017
```

Comment:
The 2015 ones are still recommended for MiniAOD: MC 80X_mcRun2_asymptotic_2016_TrancheIV_v6, Data 03Feb2017 samples. For everything else the 2017v2's are recommended. An overview of this can be found here:
https://twiki.cern.ch/twiki/bin/viewauth/CMS/TauIDRecommendation13TeV#Isolation

Reply:
We typically don't use the 80X_mcRun2_asymptotic_2016_TrancheIV_v6 and 03Feb2017 anymore though (they are replaced now by the 94X reprocessings using miniAODv3). Other parts of the current master branch won't even work properly for these samples. So it would make sense to only use 2017v2.